File: pinyougou-manager-web/src/main/webapp/js/service/itemService.js
// 服务层
pyg.service('itemService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('../item/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('../item/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('../item/findOne.do?id=' + id);
    };

    // 增加
    this.add = function (entity) {
        return $http.post('../item/add.do', entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('../item/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../item/batcheDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../item/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});

File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/ItemService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbItem;
import vo.PageResult;


/**
 * 服务层接口
 */
public interface ItemService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbItem> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbItem item, int pageNum, int pageSize);


    /**
     * 添加
     */
    void add(TbItem item);


    /**
     * 修改
     */
    void update(TbItem item);


    /**
     * 根据 ID 获取实体
     *
     * @param id
     * @return
     */
    TbItem findOne(long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);

}
